server/src/main/java/org/opensearch/index/query/IntervalsSourceProvider.java

```diff
@@ -687,24 +688,35 @@ String getUseField() {
         private final int flags;
         private final String useField;
         private final Integer maxExpansions;
+        private final boolean caseInsensitive;
 
-        public Regexp(String pattern, int flags, String useField, Integer maxExpansions) {
+        public Regexp(String pattern, int flags, String useField, Integer maxExpansions, boolean caseInsensitive) {
```

Comment:
Can we add a javadoc here that references Lucene's (one and only; lol) [matching flag](https://github.com/apache/lucene/blob/d47ff38d703c6b5da1ef9c774ccda201fd682b8d/lucene/core/src/java/org/apache/lucene/util/automaton/RegExp.java#L413). 

FWIW, I hate the C style flag lucene uses here. Maybe this is a good first lucene commit for someone to simplify the matching flags of RegExp.java?

Reply:
@nknize sure, I could work on a followup PR to lucene